src/cript/api/api.py

```diff
@@ -207,6 +208,12 @@ def _get_s3_client(self) -> boto3.client:  # type: ignore
 
         return s3_client
 
+    @property
+    def _s3_client(self):
+        if self.__s3_client is None:
+            self.__s3_client = self._get_s3_client()
+        return self.__s3_client
```

Comment:
actually the more I think about it, it's not bad because it makes the constructor lighter and this is only initialized when needed. I would like this fully encapsulated into the API, and you have already done that with the underscore prefix. 

I wonder if we need to give good errors for this or not because if this fails out of nowhere the user might be sitting there trying to track down what happened, and AWS boto errors are generally bad, but I think it is mostly fine.

Reply:
Yes, it was broken.

You commented it out entirely, which resulted in `File` nodes being broken.
(It tried to auto upload, but `_s3_client` was `None`.)

So, I needed to find a fix.
And this delayed init, allows us to not break if the `s3` is not used (for example during tests).

But it will automatically work when needed. (Integration tests.)

Reply:
I add what, I can, but it isn't great.